rust/generated_parser/src/ast_builder.rs

```diff
@@ -1627,27 +1627,49 @@ impl<'alloc> AstBuilder<'alloc> {
         &self,
         kind: arena::Box<'alloc, VariableDeclarationKind>,
         declarators: arena::Box<'alloc, arena::Vec<'alloc, VariableDeclarator<'alloc>>>,
-    ) -> arena::Box<'alloc, Statement<'alloc>> {
-        self.alloc(Statement::VariableDeclarationStatement(
+    ) -> Result<'alloc, arena::Box<'alloc, Statement<'alloc>>> {
```

Comment:
I think this currently requires the method to be added to the `FALLIBLE_METHODS` list in emit/rust.py. I'll just add it when merging this PR.

Reply:
It compiles fine, due to `TryIntoStack` (added in commit 4efb84d). `FALLIBLE_METHODS` is still needed, but only for methods that are called, and the result is *not* immediately pushed to the LR stack. Probably most of the methods listed there could be removed...